### PR TITLE
IsCanonicalVersion bool? -> bool

### PR DESCRIPTION
**Pull request by yufeih**
`bool` is enough to represent `IsCanonicalVersion`.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6441)

### CODE REVIEW

**Comment by yufeih** [comment]
diffs are expected

File: src/docfx/validation/MetadataValidator.cs [hunk 1]
```diff
@@ -52,14 +52,8 @@ public void ValidateMetadata(ErrorBuilder errors, JObject metadata, Document fil
 
             foreach (var schemaValidator in _schemaValidators)
             {
-                if (file.PageType != null)
+                if (file.PageType != null && file.PageType != "toc")
```
Comment (OsmondJiang) [inline src/docfx/validation/MetadataValidator.cs:55]: remove this hard code, every valdiation rules should have contentTypes

**Review by OsmondJiang — APPROVED** [review]

File: src/docfx/validation/MetadataValidator.cs (hunk 1 above)
Comment (yufeih) [reply src/docfx/validation/MetadataValidator.cs]: done

**Comment by OsmondJiang** [comment]
the change in sql-docs-pr is strange

**Comment by yufeih** [comment]
> the change in sql-docs-pr is strange, that's something should not be reported for reference page

Gosh, content validation rules uses `exclusions` while `metadata` rules uses `inclusions`. 

image

**Comment by OsmondJiang** [comment]
> > the change in sql-docs-pr is strange, that's something should not be reported for reference page
> 
> Gosh, content validation rules uses `exclusions` while `metadata` rules uses `inclusions`.
> 
> image

Hmmm. yes, it's caused by some history reason

**Review by OsmondJiang — APPROVED** [review]

**Pull request merged**